### PR TITLE
Use correct eos id when masking out

### PR DESCRIPTION
**Pull request by LWprogramming**

### CODE REVIEW

**Comment by LWprogramming** [comment]
By the way, why is `keep_eos` hardcoded as `False` in some places here? I searched through the blame and it used to use `include_eos_in_output` but it seems like there was something about maybe the hierarchical part [here](https://github.com/lucidrains/audiolm-pytorch/commit/65d9c407bf30e9158beda741d469e90ab8cb8179#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R197)

**Comment by lucidrains** [comment]
@LWprogramming thank you! 🙏 

so i used to keep the eos, and then taught the subsequent transformers to ignore the eos token. but it got to be too complicated, as i had to make sure the eos id is consistent across all wrappers, and do a bunch of self attention masking. so i redesigned it to simply remove the eos altogether
